qiskit/providers/aer/utils/noise_transformation.py

```diff
@@ -74,25 +74,32 @@ def approximate_quantum_error(error, *,
         ``'clifford'``.
         For further information see :meth:`NoiseTransformer.named_operators`.
     """
+    if isinstance(error, tuple) and isinstance(error[0], numpy.ndarray):
+        error = list(error)
 
-    if not isinstance(error, QuantumError):
-        error = QuantumError(error)
-    if error.number_of_qubits > 2:
+    if isinstance(error, (QuantumError, list)) or \
+            (isinstance(error, tuple) and isinstance(error[0], list)):
```

Comment:
These two checks look a bit weird - I'm not sure I understand the logic properly.  I feel like the same thing would be achieved (and be clearer, assuming I'm correct) if the final `isinstance` check was `isinstance(error[0], (list, np.ndarray))`?

Reply:
The `Kraus` channel accepts `list of np.ndarray` (ordinary Kraus) or `tuple of list of np.ndarray` (generalized Kraus), so the original interface should not allow `tuple of np.ndarray`, however, it allows for some reason. So I decided to convert `tuple of np.ndarray` to `list of np.ndarray` at the first `if`, and then check if the input type is valid at the second `if`. Should I comment that in the code?